### PR TITLE
docs: Update v4.1-upgrade.md

### PR DESCRIPTION
**Pull request by anilCSE**
## Description

Update `v4.1.2` version in upgrade docs

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by aleem1314** [comment]
Hmm. Not sure why we are getting commit hash for the local build, but goreleaser binary is showing `4.1.2` version.

**Comment by anilCSE** [comment]
> Hmm. Not sure why we are getting commit hash for the local build, but goreleaser binary is showing `4.1.2` version.

We should probably change the version to 
```
VERSION := $(shell echo $(shell git describe --tags) | sed 's/^v//')
```

File: docs/validators/upgrades/v4.1-upgrade.md [hunk 1]
```diff
@@ -64,16 +64,15 @@ cp ./build/regen $HOME/.regen/cosmovisor/upgrades/v4.1.0/bin
 Ensure the right `regen` binary has been placed in the directory:
 
 ```bash
-$HOME/.regen/cosmovisor/upgrades/v4.1.0/bin
+$HOME/.regen/cosmovisor/upgrades/v4.1.0/bin/version
```
Comment (kaustubhkapatral) [inline docs/validators/upgrades/v4.1-upgrade.md:67]:
```suggestion
$HOME/.regen/cosmovisor/upgrades/v4.1.0/bin/regen version
```

Comment (kaustubhkapatral) [inline docs/validators/upgrades/v4.1-upgrade.md:67]:
```suggestion
$HOME/.regen/cosmovisor/upgrades/v4.1.0/bin/regen version
```

**Review by kaustubhkapatral — CHANGES_REQUESTED** [review]

**Review by kaustubhkapatral — APPROVED** [review]

**Review by aleem1314 — APPROVED** [review]

**Comment by ryanchristo** [comment]
@mergifyio backport release/v4.1.x

**Pull request merged**